spatialdata: add spatialdata package to spack

downloadico opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

downloadico committed:
spatialdata: add spatialdata package to spack

downloadico committed:
spatialdata: fixed style problems

becker33 commented on var/spack/repos/builtin/packages/spatialdata/package.py:
```diff
@@ -0,0 +1,53 @@
+
+
+class Spatialdata(AutotoolsPackage):
+    """Spatialdata is a C++ library for
+
+    interpolating values for spatially distributed data,
+    converting coordinates among geographic projections using Proj,
+    nondimensionalization of quantities,
+    specification of units via Pyre (optional).
+
+    This library is used in the finite-element code PyLith
+    (https://github.com/geodynamics/pylith). The primary focus is specification
+    of parameters that vary in space, such as values for boundary conditions
+    and parameters of constitutive models. This provides a specification of
+    these parameters independent of the discretization."""
```
Can you edit this so that there is a <100 character single-line summary on the first line, and the rest follows?

becker33 commented on var/spack/repos/builtin/packages/spatialdata/package.py:
```diff
@@ -0,0 +1,53 @@
+
+    version(
+        "develop",
+        git="https://github.com/geodynamics/spatialdata/",
```
Can you move the `git` attribute to be class scoped?

```
url = "..."
git = "..."

license(...)

version("develop", branch="develop", submodules="true")
```

That way users will be able to use `spack isntall spacialdata@git.REF` to install from other refs in the git repo. It's also generally best practice to have the `preferred` version not be a git branch, is there a reason we should do otherwise specifically for this package?

downloadico replied:
Thanks!  Done.

downloadico committed:
spatialdata: fixed issues from review and changed autoreconf stage.
added short description of package.
incorporated changes from reviewer.
added a resource for the CIG M4 macros.
added dependencies on python, and py-setuptools.
removed dependency on catch2 as it is only needed for testing.
made autoreconf run autoupdate then autoreconf each with the using the CIG M4
macros.

kwryankrattiger commented on var/spack/repos/builtin/packages/spatialdata/package.py:
```diff
@@ -0,0 +1,63 @@
+    depends_on("python")
+    depends_on("py-setuptools")
+    depends_on("py-cig-pythia")
+    depends_on("proj")
```
I want to say proj had an API change between a couple of versions. Is there a bounds required here?

downloadico replied:
https://github.com/geodynamics/spatialdata says that Proj version 6.0 or later is required.  I'll add that

kwryankrattiger commented on var/spack/repos/builtin/packages/spatialdata/package.py:
```diff
@@ -0,0 +1,63 @@
+    version("3.1.0", sha256="dd6caccbf41a51928183d6a1caf2380aa0ed0f2c8c71ecc9b2cd9e3f23aa418c")
+
+    # M4 macros shared for the CIG codes
+    resource(name="autoconf_cig", git="https://github.com/geodynamics/autoconf_cig.git")
```
I think this should probably have a specific ref./checksum, otherwise it makes it difficult to reliably reproduce the build.

downloadico replied:
I set it to the version they point to in their github page.

kwryankrattiger reviewed (commented):
A couple of questions/comments.

downloadico committed:
spatialdata: specified commit for the autoconf_cig resource. set minimum version of Proj to 6.

kwryankrattiger commented on var/spack/repos/builtin/packages/spatialdata/package.py:
```diff
@@ -0,0 +1,67 @@
+    depends_on("python")
+    depends_on("py-setuptools")
+    depends_on("py-cig-pythia")
+    depends_on("proj@:6")
```
This says 6 or older, use`@6:`

downloadico replied:
D'oh!

downloadico committed:
spatialdata: fixed version of Proj required.
